csrc/macro.h

```diff
@@ -0,0 +1,21 @@
+#define NVRTC_SAFE_CALL(x)                                        \
+  do {                                                            \
+    nvrtcResult _result = x;                                       \
+    if (_result != NVRTC_SUCCESS) {                                \
+      std::cerr << "\nerror: " #x " failed with error "           \
+                << nvrtcGetErrorString(_result) << '\n';           \
+      exit(1);                                                    \
+    }                                                             \
+  } while(0)
+
+#define CUDA_SAFE_CALL(x)                                         \
```

Comment:
We have runtime and driver calls, so we should probably need this for the runtime API as well.